cli/ops/tty.rs

```diff
@@ -250,3 +251,113 @@ pub fn op_isatty(
     })?;
   Ok(JsonOp::Sync(json!(isatty)))
 }
+
+#[derive(Serialize)]
+struct ConsoleSize {
+  columns: u32,
+  rows: u32,
+}
+
+#[cfg(unix)]
+fn console_size(rid: i32) -> Result<ConsoleSize, std::io::Error> {
+  use libc::{ioctl, winsize, TIOCGWINSZ};
+  use std::mem::MaybeUninit;
+
+  let mut count: u32 = 0;
+  loop {
```

Comment:
What's the purpose of this loop? I think this is unnecessary. ioctl should be called exactly once.

Reply:
There seem to be a chance that ioctl doesn't return the right answer the first time. Every implementation I found uses a loop, here's the code from Node.JS:
```js
  do
    err = ioctl(uv__stream_fd(tty), TIOCGWINSZ, &ws);
  while (err == -1 && errno == EINTR);
```

Reply:
Interruption of a system call can happen when it is interrupted by signal handler, but I don't see a reason to special case this one system call when all others are just left alone.

If this is something we care about then automatic system call restarts are probably the way to go.

See the "Interruption of system calls and library functions by signal handlers" section in signal(7): http://man7.org/linux/man-pages/man7/signal.7.html